### PR TITLE
[PT-D][3/N] Sync TP API change to Pytorch

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89535

Differential Revision: [D41497546](https://our.internmc.facebook.com/intern/diff/D41497546)

### CODE REVIEW

File: torch/distributed/_tensor/parallel/style.py [hunk 1]
```diff
@@ -181,15 +213,19 @@ def make_output_tensor(
...
+
     Args:
-        output (DTensor): output of module to be converted.
-        device_mesh (Optional[DeviceMesh]): :class:``DeviceMesh`` object needed to
-            replicate the output and it needs to be a 1D ``device_mesh`` and we will
-            throw exceptions if a non-1D ``device_mesh`` is passed in. If no
-            ``device_mesh`` is passed in, we will reuse the one from output.
+        output (:class:`DTensor`):
+            Output of module to be converted.
+        device_mesh (DeviceMesh, optional):
```
Comment (wanchaol) [inline torch/distributed/_tensor/parallel/style.py:220]: nit: :class:`DeviceMesh`

**Review by wanchaol — APPROVED** [review]

File: torch/distributed/_tensor/parallel/style.py (hunk 1 above)
Comment (fduwjj) [reply torch/distributed/_tensor/parallel/style.py:220]: Opps, let me sync this.

**Comment by fduwjj** [comment]
Manually rebase and addressed the comment.

**Pull request closed without merging**